convert2rhel/unit_tests/utils_test.py

```diff
@@ -874,7 +888,312 @@ def communicate(self):
         pass
 
     def poll(self):
-        return 0
+        return self.return_code
+
+    def close(self):
+        pass
+
+
+@pytest.mark.parametrize(
+    ("output", "skip_stdout_decode", "expected"),
+    (
+        (DummyPopenOutput([u"test".encode()]), True, b"test"),
+        (DummyPopenOutput([u"testê café".encode("utf-8")]), False, u"testê café"),
+        (DummyPopenOutput([u"testê café".encode("utf-8")]), False, u"testê café"),
+        (DummyPopenOutput([u"test".encode()]), False, "test"),
```

Comment:
```suggestion
        (DummyPopenOutput([u"test".encode("ascii")]), True, b"test"),
        (DummyPopenOutput([u"test".encode("ascii")]), False, u"test"),
        (DummyPopenOutput([u"testê café".encode("utf-8")]), True, u"testê café".encode("utf-8")),
        (DummyPopenOutput([u"testê café".encode("utf-8")]), False, u"testê café"),
```
* Fix one of the test case's data to test non-ascii data with skip_stdout_decode=True
* Be sure to specify `u` on the ascii-only string when skip_stdout_decode=False.
* Order the output as, contans ascii-only data, non-ascii-only data.
* Always encode() using an explicit encoding. (Note: the encoding calls I added an encoding too could have used utf-8 but it seemed to be clearer what the purpose of the test was to use ascii as the encoding.

Reply:
Not resolving this comment, but we have this mapped under the gist mentioned in a comment here, so should be fine to resolve it anyway.